flytekit/sdk/tasks.py

```diff
@@ -406,12 +406,16 @@ def spark_task(
         cache_version='',
         retries=0,
         interruptible=None,
+        inputs=None,
```

Comment:
Is this handleded differently as compared to @inputs decorator? If so we should document that correctly, as this is very confusing.

ALSO at the moment I think only a few input types are supported like primitives and blobs right?

Reply:
Moved to use the inputs decorator. Added the check for primitives for now

Reply:
I wouldn't mix the decorator with the generic usage. The decorator patterns are part of the SDK's basic python programming model so the assumption is that they will always be python. It's easy enough to expose a helper function or class for Scala spark jobs elsewhere.

Reply:
(if you want to make available through this file, that's fine too. I just think mixing a decorator and common object definitions will eventually cause a problem)

Reply:
and continuing Ketan's thought, I would also just avoid adding the input arg that way all together. Especially since as your test below shows, you use the input decorator as a helper function--which is a bit confusing. Better to use a new interface and just a normal `{'a': Types.Integer}` style annotation

Reply:
I originally wasn't using the decorator but changed after Ketan's comment to be in-sync with how we do this for Presto as well : https://github.com/lyft/flytekit/blob/master/tests/flytekit/common/workflows/presto.py#L11

Reply:
Updated to stop using the existing spark_task decorator. I do think we should make the generic_spark_task available from this file as well. Added it separately as a helper function.